truenas_install/__main__.py

```diff
@@ -235,17 +246,62 @@ def configure_system_for_zectl(boot_pool):
         run_command(["zfs", "set", "org.zectl:bootloader=grub", root_ds])
 
 
+def read_license(root):
+    license_path = os.path.join(root, "data/license")
+    try:
+        with open(license_path) as f:
+            return License.load(f.read().strip('\n'))
+    except Exception as e:
+        return None
+
+
+def precheck(old_root):
+    if old_root is not None:
+        if licenseobj := read_license(old_root):
+            if ContractType(licenseobj.contract_type) in [ContractType.silver, ContractType.gold]:
+                if licenseobj.contract_end > datetime.utcnow().date():
+                    db_path = database_path(old_root)
+                    if os.path.exists(db_path):
+                        try:
+                            if query_row(
+                                "SELECT * FROM services_services WHERE srv_service = 's3' AND srv_enable = 1",
+                                db_path,
+                            ) is not None:
+                                return (
+                                    "You have built-in S3 service enabled in your configuration. This is not supported "
+                                    "anymore. Please, contact our support team."
+                                )
+                        except Exception:
+                            pass
+
+                    for p in psutil.process_iter():
+                        if p.name() == "minio":
+                            # FIXME: ignore if minio is running in k3s
```

Comment:
We'll need to solve this before merge. The process is going to be that the customer will start the minio app alongside the host S3 service and add the app as a node to their existing cluster and minio will transfer the data natively.

Then the host service will be shut-off and the app will continue to run so I predict this will fail 100% of the time?

Reply:
I would be super glad to do it, but apps are broken in Cobia. Can we create a ticket instead?

Reply:
@sonicaj sorry, something was off locally, they are working after a reboot